### DISCUSSION TITLE
How to make confirmation modal before create in CreatePage

### DISCUSSION THREAD

**Discussion by NaungYeHtet** [filamentphp/filament#9537] [Help] [answered]
```
protected function beforeCreate(): void
{
    Notification::make()
        ->warning()
        ->title('Attention')
        ->body('This will notify to all user!')
        ->persistent()
        ->actions([
            Action::make('Proceed')
                ->button()
                ->action(fn () => $this->proceed()),
            Action::make('Cancel')
                ->button()
                ->action(fn () => $this->halt()),
        ])
        ->send();
}
```

**Comment by NaungYeHtet** [comment, marked as answer]
```
protected function getCreateFormAction(): Action
{
    return Action::make('create')
        ->label(__('filament-panels::resources/pages/create-record.form.actions.create.label'))
        ->requiresConfirmation()
        ->action(fn () => $this->create())
        ->keyBindings(['mod+s']);
}
```

**Reply by pabbec** [reply]
This works fine, but when you upload a file, the button doesn't change state (% Uploading file) or become disabled until the upload is complete. Do you know if there is a way to fix this?

**Comment by LeamonaDor1** [comment]
And what if I want a validation check on the fields to be performed before the `requiresConfirmation`?
I liked this less:

![image](https://github.com/user-attachments/assets/cf10f90d-1d20-4980-83f8-8d832fde9cd1)

**Reply by leandrocfe** [reply]
use this https://x.com/leandrocfe/status/1800169819880472924

**Comment by LeamonaDor1** [comment]
I didn't explain myself enough.
I have two reasons why I want validation to occur before a confirmation message pops up.
1. I want the user to fill out the form.
 My code will have `requiresConfirmation` with `if` in it. 
Only if a certain field equals so will a confirmation message appear.
2. If a certain field is `null` even though it is required there is no reason to pop up a message: "Are you sure?". Too bad. It's better to stop a step earlier and warn him about it.

**Comment by LuizCristino** [comment]
@LeamonaDor1 I don't know if you are still interested in some solution for this discussion, but I had to do something similar to what you wanted. 

I resolved using this approach.

```php
// This example is for the EditRecord page.
protected function getSaveFormAction(): Action
    {
        return Action::make('save')
            ->label(__('filament-panels::resources/pages/edit-record.form.actions.save.label'))
            ->requiresConfirmation(function () {
                try {
                    // This method launches the validation and throws a error.
                    $this->form->getState();
                } catch (\Exception $e) {
                    // So if a error is thrown you do not want a confirmation, so the user receives de error feedback because of the action
                    return false;
                }

                // FOrm is valid, proceed to request the user confirmation
                return true;
            })
            ->action(fn () => $this->save())
            ->keyBindings(['mod+s']);
    }
```
